### DISCUSSION TITLE
Changing Default JsonSerilizerOptions

### DISCUSSION THREAD

**Discussion by HarCass** [elsa-workflows/elsa-core#6234] [Q&A]
Hello, I'm currently using Elsa 3's server with the studio and I'm wondering if there is a way to change the default JsonSerializerOptions that Elsa uses in its HttpResponse activities, I need it to ignore when writing null values and use camel case. I couldn't seem to find a way to apply this globally within Elsa either in code or studio. At the moment I am adding attributes to my objects to achieve what I want, however this is very long winded and I was wondering if I was just missing some easy way to do this.

I couldn't find anything in the docs about this or a discussion about it, I have tried something like this to no avail
```cs
builder.Services
    .AddElsa(elsa =>
    {
        elsa.Services.ConfigureHttpJsonOptions(options =>
        {
            options.SerializerOptions.DefaultIgnoreCondition = JsonIgnoreCondition.WhenWritingNull;
            options.SerializerOptions.PropertyNamingPolicy = JsonNamingPolicy.CamelCase;
        });
    });
```

Any help is appreciated, thanks in advance!

**Comment by sfmskywalker** [comment]
`ConfigureHttpJsonOptions` configures ASP.NET Core's HTTP JSON options, but it does not control Elsa's **Write HTTP Response** activity. In Elsa 3.0.0, the activity's JSON content factory calls `JsonSerializer.Serialize(content)` without passing any configured options, so there is no supported global option for camel-case naming or null omission on that activity.

For that version, the practical workaround is to serialize the response body yourself with the `JsonSerializerOptions` you need and pass the resulting string to **Write HTTP Response**. If you need this everywhere, replace Elsa's `IHttpContentFactory` for `application/json` with your own implementation that serializes with those options.

Relevant source: [`JsonContentFactory.CreateHttpContent`](https://github.com/elsa-workflows/elsa-core/blob/3.0.0/src/modules/Elsa.Http/ContentWriters/JsonContentFactory.cs#L16-L24).
